airbyte-config/init/build.gradle

```diff
@@ -32,10 +32,18 @@ tasks.named("buildDockerImage") {
 
 Task publishArtifactsTask = getPublishArtifactsTask("$rootProject.ext.version", project)
 
+
+task downloadCombinedConnectorCatalog(type: JavaExec, dependsOn: compileJava) {
+    classpath = sourceSets.main.runtimeClasspath
+    mainClass = 'io.airbyte.config.init.CombinedConnectorCatalogDownloader'
+}
+
+
 /**
  * Simple task that checks if all icons in the seed definition files exist as well as
  * that no icon in the icons folder is unused.
  */
+// TODO (ben): Refactor this in a future PR
 class IconValidationTask extends DefaultTask {
```

Comment:
@airbytehq/connector-operations In preparation for the monorepo switch, do we know who is responsible for validation of icons? Its the OSS side correct?

Reply:
I think since (at the bottom) we've got `check.dependsOn validateIcons`, I believe that both the OSS and Cloud builds will run this task

Reply:
If this is going to run on the platform side, we shouldn't reference any definition yamls since they will not be available. Should probably use the JSON catalog to get which icons we need to check.

This is in a weird spot right now because the images are bundled with the platform - this should obviously move to be a sole concern of the connectors side later. For now, I think it makes sense for this to run platform side.

Reply:
Ok great.

Just refactored it into a task that uses the LocalProvider which references the local copy of oss_catalog from gcs